apps/chromealive-core/lib/bridges/BridgeToDevtoolsPrivate.ts

```diff
@@ -0,0 +1,134 @@
+            const payload = restOfMessage.payload;
+            if (payload.event === '${MessageEventType.OpenSelectorGeneratorPanel}') {
+              (${openSelectorGeneratorPanel.toString()})(DevToolsAPI, '${extensionId}');
+            } else {
+              console.log('UNHANDLED MESSAGE FROM CORE: ', destLocation, responseCode, payload);
+            }
+          };
+          (${interceptElementOverlayDispatches.toString()})('${___sendToCore}', '${MessageEventType.OverlayDispatched}');     
+        })();`,
+      }),
+      devtoolsSession.send('Runtime.runIfWaitingForDebugger'),
+    ]).catch(() => null);
+  }
+
+  public close() {
+    this.devtoolsSessionMap = new Map();
+  }
+
+  public send(message: any) {
+    const [destLocation, responseCode, restOfMessage] = extractStringifiedComponentsFromMessage(message);
+    const devtoolsSessionEntry = this.devtoolsSessionMap.entries().next().value;
+    if (!devtoolsSessionEntry) return;
+
+    const [devtoolsSession, contextIds] = devtoolsSessionEntry;
+    const contextId = contextIds[0];
+    return this.runInBrowser(
+      devtoolsSession,
+      contextId,
+      `window.${___receiveFromCore}('${destLocation}', '${responseCode}', ${restOfMessage});`
+    );
+  }
+
+  private runInBrowser(devtoolsSession: IDevtoolsSession, contextId: number, expressionToRun: string) {
+    const response = devtoolsSession.send('Runtime.evaluate', {
+      expression: expressionToRun,
+      contextId,
+      awaitPromise: false,
+      returnByValue: false,
+    });
+    response.catch(err => {
+      if (err instanceof CanceledPromiseError) return;
+      throw err;
+    });
+  }
+
+  private handleIncomingMessageFromBrowser(event: any) {
+    if (event.name !== ___sendToCore) return;
+    const [destLocation] = extractStringifiedComponentsFromMessage(event.payload);
+    this.emit('message', event.payload, { destLocation });
+  }
+
+  private onContextCreated(devtoolsSession: IDevtoolsSession, event: Protocol.Runtime.ExecutionContextCreatedEvent): void {
+    if (!this.devtoolsSessionMap.has(devtoolsSession)) {
+      this.devtoolsSessionMap.set(devtoolsSession, new Set());
+    }
+    this.devtoolsSessionMap.get(devtoolsSession).add(event.context.id);
+  }
+
+  private onContextDestroyed(devtoolsSession: IDevtoolsSession, event: Protocol.Runtime.ExecutionContextDestroyedEvent): void {
+    const contextIds = this.devtoolsSessionMap.get(devtoolsSession);
+    if (!contextIds) return;
+
+    contextIds.delete(event.executionContextId);
+    if (!contextIds.size) {
+      this.devtoolsSessionMap.delete(devtoolsSession)
+    }
+  }
+
+  private onContextCleared(devtoolsSession: IDevtoolsSession): void {
+    this.devtoolsSessionMap.delete(devtoolsSession);
+  }
+}
+
+// METHODS TO RUN IN BROWSER CONTEXT ////////////////////////////////////////////////////////
+
+// eslint-disable-next-line @typescript-eslint/no-shadow
+function openSelectorGeneratorPanel(DevToolsAPI: any, extensionId: string) {
```

Comment:
Not sure this needs a change in this pr... but I don't think I like the idea of having functions defined in node world that you stringify on sending into the client. It might be clearer to assign the toString of the fn to a const variable here, or have these live in another file

Reply:
Agreed.